plans/objchange/compatible.go

```diff
@@ -219,8 +218,10 @@ func assertValueCompatible(planned, actual cty.Value, path cty.Path) []error {
 		// Anything goes, then
 		return errs
 	}
-	if problems := planned.Type().TestConformance(actual.Type()); len(problems) > 0 {
-		errs = append(errs, path.NewErrorf("wrong final value type: %s", convert.MismatchMessage(actual.Type(), planned.Type())))
+	if problems := actual.Type().TestConformance(planned.Type()); len(problems) > 0 {
```

Comment:
From the docs of `TestConformance`:
```
// TestConformance recursively walks the receiver and the given other type and
// returns nil if the receiver *conforms* to the given type.
```
According to my understanding of the docs and the function definition of 
`func (t Type) TestConformance(other Type) []error`,
 the function should be called on the *actual* type and the parameter `other` should be the *planned* type.

Reply:
That would be my reading of the code as well. Since this is unchanged from the initial commit, I could definitely see how this could be a typo because the arguments are otherwise always handled in the actual->planned order. 

Do you have a test we could add that highlights this change?